RealmSwift/SwiftUI.swift

```diff
@@ -603,7 +603,7 @@ extension Projection: _ObservedResultsValue { }
     public mutating func update() {
         // When the view updates, it will inject the @Environment
         // into the propertyWrapper
-        if storage.configuration == nil || storage.configuration != configuration {
```

Comment:
Why did this fix it?

Reply:
If the configuration stored in `.environment(\.realmConfiguration, ...)`did not match a config that was passed to the initialiser it would override `storage.configuration` with whatever the `.environment` stored, which is likely just a default realm configuration.

Reply:
Can we just add the schema to the Equatable function?, this will probably reset the configuration every time we render the view.

Reply:
Good point Diana.

Reply:
Yep, you're right

Reply:
Is there any way for the configuration from the Environment to change between calls to `update()` without a new View being constructed? The documentation implies it's possible (it says that `update()` is called each time the value for the environment key changes), but I don't see any way for it to actually happen.

Reply:
I can't see a way to update the value without it reinitialising the entire view (using `print(Self._printChanges())` to debug that). The docs also mention `You can use this property wrapper to read — but not set` which is kinda contradictory to `If the value changes, SwiftUI updates any parts of your view that depend on the value`. Maybe this means if you modify from the top level with something like?

```swift
    @State var changeConfig = false

    var body: some Scene {
        WindowGroup {
            ContentView(changeConfig: $changeConfig)
                .environment(\.realmConfiguration, changeConfig ? Realm.Configuration(inMemoryIdentifier: "foo") : Realm.Configuration(inMemoryIdentifier: "bar"))
        }
    }
```